src/kernels/helpers.ts

```diff
@@ -294,10 +294,10 @@ export function compareKernels(
         b.kernelSpec.metadata?.vscode?.originalDisplayName || b.kernelSpec?.display_name || '';
     const kernelSpecNameOfA = originalSpecFileA
         ? path.basename(path.dirname(originalSpecFileA))
-        : a.kernelSpec?.display_name || '';
+        : a.kernelSpec?.name || '';
```

Comment:
Ran into this while doing my quick dive into compareKernel. I don't have a specific repro here, but from the usage looks like this could have been causing preferred kernel selection issues.

Reply:
Yes, looks like an obvious issue.